### ISSUE TITLE
Error using role in a query

### ISSUE THREAD

**Issue by Marcorus** [Seddryck/NBi#187]
Hello,

I have an issue using Role to a query like describe here : http://www.nbi.io/docs/connection-roles/
When defining it like this in the nbits file i have the following error : 
`<query roles="My SSAS ROLE">`

> NBi.NUnit.Runtime.TestSuite.Verify if query equal itself:
> System.ArgumentException : The '
>             ;Roles' property name is not formatted correctly.

When i use it has described in the doc, i have the ref attribute error that is not declared : 
`<query ref="OlapQA" roles="My SSAS ROLE">`

> NBi.NUnit.Runtime.TestSuite.ExecuteTestCases:
> System.ArgumentException : The test suite is not valid. Check with the XSD. 1 error has been found during the validation of the test-suite:
>     At line 35: The 'ref' attribute is not declared.

What am I missing ? Is there still two parameters needed ?  What is the ref attribute ?

**Comment by Seddryck** [comment]
About `roles` ... is there something special in your role's name (spaces, any whitespaces, comma, brackets, quotes ...)?

About the attribute `ref`, it's clearly optional but if NBi complains about it, it's another bug. The goal of `ref` is to reference a query and avoid to repeat the same query in many tests.

**Comment by Marcorus** [comment]
Yes there is a space in my role. 

Y tried to run it without space just to verify and still the same problem.
`<query roles="MySSASROLE">`

Error :

> NBi.NUnit.Runtime.TestSuite.Verify if query equal itself:
> System.ArgumentException : The '
>             ;Roles' property name is not formatted correctly.

**Comment by Seddryck** [comment]
Huuum, I think that the problem is somewhere else. It sounds that you have a carriage return somewhere in your connection-string. Could you be sure to have the connection-string and the role attributes on the same line?

``` xml
<query connectionString="..." roles="MySSASROLE">
```

and not something similar to

``` xml
<query connectionString="...
" 
   roles="MySSASROLE"
>
```

**Comment by Marcorus** [comment]
It worked.

I was using `<default apply-to="system-under-test">` for this query. So I created a connection string in the settings part and change the query to `<query connectionString="@OLAP" roles="My Role">` and it worked.

But no carriage return found ( Checked with notepad++)

Thanks for the support and the good work on this framework.

**Comment by Seddryck** [comment]
I'll let this open and take a look further, it should also work with a `default`. 

Thx for the feedback and feel free to report any issue.

**Comment by Seddryck** [comment]
I found a way to reproduce and fixed (anyway not sure it was really the root cause in your case).

preview available at https://ci.appveyor.com/project/CdricLCharlier/nbi/build/1.14.0-ci.51/artifacts

**Comment by Seddryck** [comment]
Included in release v1.14
